controllers/infrastructure/byomachine_controller.go

```diff
@@ -414,6 +443,59 @@ func (r *ByoMachineReconciler) setPausedConditionForByoHost(ctx context.Context,
 	return helper.Patch(ctx, machineScope.ByoHost)
 }
 
+func (r *ByoMachineReconciler) setHostInfoForByoMachine(ctx context.Context, machineScope *byoMachineScope) error {
+	helper, err := patch.NewHelper(machineScope.ByoMachine, r.Client)
+	if err != nil {
+		return err
+	}
+
+	machineScope.ByoMachine.Status.HostInfo = machineScope.ByoHost.Status.HostDetails
+
+	return helper.Patch(ctx, machineScope.ByoMachine)
+}
+
+func (r *ByoMachineReconciler) setInstallationSecretForByoHost(ctx context.Context, machineScope *byoMachineScope) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithValues("cluster", machineScope.Cluster.Name)
+	installerConfig, ready, err := r.getInstallerConfigAndStatus(ctx, machineScope)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if !ready {
+		logger.Info("Installer config is not ready, requeuing")
+		return ctrl.Result{RequeueAfter: RequeueForInstallerConfig}, errors.New("installer secret is not yet ready")
```

Comment:
we need to return nil as it will requeue after `RequeueForInstallerConfig` duration

Reply:
We need the error because caller of that function returns only if err != nil. Similar logic is in attachByoHost()  if "No hosts found"  they requeue after a specific time

https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/controllers/infrastructure/byomachine_controller.go#L445-L450

Reply:
Added the additional check for RequeueAfter along with err in the caller as suggested 👍